src/test/java/htsjdk/samtools/util/HttpUtilsTest.java

```diff
@@ -0,0 +1,35 @@
+package htsjdk.samtools.util;
+import java.io.IOException;
+import java.net.URL;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import htsjdk.HtsjdkTest;
+
+public class HttpUtilsTest extends HtsjdkTest {
+    @DataProvider(name = "existing_urls")
+    public Object[][] testExistingURLsData() {
+        return new Object[][]{
+                {"http://broadinstitute.github.io/picard/testdata/index_test.bam"}
+        };
+    }
+    
+    @Test(dataProvider="existing_urls")
+    public void testGetHeaderField(final String url) throws IOException {
+    final String field = HttpUtils.getHeaderField(
+            new URL(url),
+            "Content-Length"
+            );
+    Assert.assertNotNull(field);
+    final long length = Long.parseLong(field);
+    Assert.assertTrue(length>0L);
+    }
+    
+    @Test(dataProvider="existing_urls")
+    public void testGetETag(final String url) throws IOException {
+    final String field = HttpUtils.getETag(new URL(url));
```

Comment:
Is this a useful API to keep? It's not used in picard or htsjdk or gatk. It would be better to remove/deprecate `getEtag()` and `resourceAvailable` than add tests for them.

Reply:
You never know who's using something downstream.  I think we're hoping to deprecate the entire http code package in the future once we've tested out an http path replacement, but I'm not sure if it's worth deprecating bits now.